### PR TITLE
feat(load): add a javascript load queue

### PR DESCRIPTION
**Pull request by cubesky**
**What kind of change does this PR introduce?** (check one with "x")

- [ ] Bug fix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

____

**Description**

Add a javascript load queue.   
You can use `queue.offer(function(){ });` to add a function to load queue and it will automatic called when document DOM finished loading.   
You can also use this in `head.yml` to use this queue.    
Now, Bing background and Comment (Disque/Duoshuo) loading using this load queue. 

____

**Verification steps**

No verification steps.

### CODE REVIEW

**Comment by cubesky** [comment]
Use lazyload for background image. Now, background image will be shown when the image finished download. This will avoid viewer looking for a picture is downloading in progress.

**Comment by Halyul** [comment]
顺带把图库的修复了啊🌚🌚

File: layout/_partial/head.ejs [hunk 1]
```diff
@@ -119,6 +121,16 @@
     </script>
     <% } %>
 
+    <!-- Bing Background -->
+    <% if(theme.background.bing.enable) { %>
+        <script type="text/javascript">
+            queue.offer(function(){
+                $('body').attr("data-original","https://api.i-meto.com/bing?<%= theme.background.bing.parameter %>");
```
Comment (pidupuis) [inline layout/_partial/head.ejs]: - Please use single quotes inside JavaScript code
- Please put a space after comma

File: source/js/queue.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    this.dataStore = [];
+    this.offer = offer;
+    this.poll = poll;
+    this.execNext = execNext;
+    this.debug = false;
+    this.startDebug = startDebug;
+
+
+    function offer(element) {
+        if(this.debug) console.log("Offered a Queued Function.");
```
Comment (pidupuis) [inline source/js/queue.js]: - Please user single quote inside JavaScript code (change it for entire file)
- Please surround `if` statement with brackets even for one line, and display it in several lines (change it for entire file)

File: source/js/queue.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    }
+
+    function poll() {
+        if(this.debug) console.log("Polled a Queued Function.");
+        return this.dataStore.shift();
+    }
+
+    function execNext() {
+        var nextfunc=this.poll();
```
Comment (pidupuis) [inline source/js/queue.js]: - Please surround `=` by space

File: source/js/queue.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        if(nextfunc != undefined) {
```
Comment (pidupuis) [inline source/js/queue.js]: - Please use strict equality

File: layout/_partial/head.ejs [hunk 5]
```diff
@@ -84,6 +84,8 @@
     <%- css("css/style.min") %>
     <%- partial("_partial/config_css") %>
     <%- js("js/jquery.min") %>
+    <%- js("js/queue") %>
```
Comment (pidupuis) [inline layout/_partial/head.ejs]: - Please use single quotes inside JavaScript code (change it for the entire stack of import)

File: layout/_partial/footer-option.ejs [hunk 6]
```diff
@@ -15,18 +15,20 @@
 <% if(theme.comment.use == "duoshuo") { %>
```
Comment (pidupuis) [inline layout/_partial/footer-option.ejs]: - Please use strict equality
- Please use single quotes in JavaScript code

**Comment by cubesky** [comment]
Ok, I will fix it as soon as possible.

**Comment by cubesky** [comment]
@pidupuis I fixed it. Please review.

**Pull request merged**